src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -1278,16 +1278,17 @@ private void doHandleSwitchExpression(JCSwitchExpression tree) {
     }
     //where:
         private boolean hasTry(JCSwitchExpression tree) {
-            boolean[] hasTry = new boolean[1];
-            new TreeScanner() {
+            var scan = new TreeScanner() {
```

Comment:
I think `scanner` might be a better name for this:
```suggestion
            var scanner = new TreeScanner() {
```

Reply:
I agree, the first guideline of the excellent guide written by Stuart Marks about the usage of 'var' is "Choose variable names that provide useful information [1].

[1] https://openjdk.org/projects/amber/guides/lvti-style-guide#G1

Reply:
To be honest, I don't see a big difference between "scan", "scanner" and "treeScanner" - these describe the type of the variable, but not its purpose. So, I've used `hasTryScanner`, which conveys the purpose as well. Not a big reason to use less readable tricks with anonymous and var either - javac normally uses local classes to perform tasks like this.

Reply:
I agree, that we should not spend too much time on that code.
At the same time, `scan.scan(...)` is not very readable, I believe that `scanner` or `treeScanner` are better names.